mlir-sycl/doc/design/SYCLGridRepresentation.md

```diff
@@ -0,0 +1,244 @@
+# SYCL Index Space Representation
+
+Identifying a work-item in the SYCL index space (using its global id or the
+combination of its work-group id and its local id within a work-group) is
+necessary to perform memory access analysis on SYCL device code, so representing
+these SYCL index space identifiers is required to enable such analysis through
+the SYCL dialect.
+
+**Note**: Through this document, we assume `memref` and `llvm.ptr` are used for
+pointer types, `llvm.struct` for structured types and `sycl.cast` is the
+operation used to perform a casting from a SYCL type to a base class.
```

Comment:
We could have a SYCL type inheriting from more than one class in an implementation, so I'd rather keep this as is

Reply:
yes but the classes are still part of its inheritance chain. Suggest:

" to a base class." -> " to its base class (or one of its base classes in case of multiple inheritance)"